core/src/main/scala/com/nvidia/spark/rapids/tool/qualification/Qualification.scala

```diff
@@ -157,37 +163,46 @@ class Qualification(outputDir: String, numRows: Int, hadoopConf: Configuration,
   private def qualifyApp(
       path: EventLogInfo,
       hadoopConf: Configuration): Unit = {
+    var message: String = ""
     try {
       val startTime = System.currentTimeMillis()
       val app = QualificationAppInfo.createApp(path, hadoopConf, pluginTypeChecker, reportSqlLevel,
         mlOpsEnabled)
       if (!app.isDefined) {
         progressBar.foreach(_.reportUnkownStatusProcess())
-        logWarning(s"No Application found that contain SQL for ${path.eventLog.toString}!")
-        None
+        message = s"No Application found that contain SQL for ${path.eventLog.toString}!"
+        logWarning(message)
+        appStatusMap.put(path, StatusNotAvailable(None, message))
       } else {
         val qualSumInfo = app.get.aggregateStats()
         if (qualSumInfo.isDefined) {
           allApps.add(qualSumInfo.get)
+          appStatusMap.put(path, StatusSuccess(app))
```

Comment:
Why do we need the entire "`app`" to be part of the `StatusSuccess`?
This causes a significant overload on the memory and more frequent `OOM`. Notice that the original code create a local `app` variable, but keeps track of the `qualSumInfo`.
`appStatusMap` will have objects that reference `app`. The Garbage collector will not be able to cleanup the memory after leaving the `qualifyApp` scope. The application will eventually crash because we keep the app, appInfo, statusInfo in memory until end of execution.

Reply:
Removed storing the entire app. Storing just the `appId` string